packages/js/src/feeds/helpers.ts

```diff
@@ -0,0 +1,263 @@
+import { ApiService } from '@novu/client';
+
+import type { NovuEventEmitter } from '../event-emitter';
+import { NotificationActionStatus, NotificationButton, NotificationStatus, TODO } from '../types';
+import { Notification } from './notification';
+import {
+  MarkNotificationActionAsArgs,
+  MarkNotificationAsArgs,
+  MarkNotificationsAsArgs,
+  RemoveNotificationArgs,
+  RemoveNotificationsArgs,
+} from './types';
+
+const getOptimisticMarkAs = (status: NotificationStatus): Partial<Notification> => {
+  switch (status) {
+    case NotificationStatus.READ:
+      return { read: true, seen: true };
+    case NotificationStatus.UNREAD:
+      return { read: false, seen: true };
+    case NotificationStatus.SEEN:
+      return { seen: true };
+    case NotificationStatus.UNSEEN:
+      return { seen: false };
```

Comment:
I didn't add that just because we haven't fixed that on the server side. Like to not confuse the users with an optimistic state different from the result (server) state.